sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GlobalAddressResolver.java

```diff
@@ -94,69 +97,142 @@ public GlobalAddressResolver(
+                                                                containerLinkToCollection.left,
+                                                                Configs.getMinConnectionPoolSizePerEndpoint()
+                                                        );
+
+                                                return this.submitOpenConnectionInternal(
+                                                        addressInformation,
+                                                        containerLinkToCollection.getRight(),
+                                                        proactiveContainerInitConfig,
+                                                        connectionsPerEndpointCountForContainer,
+                                                        hint
+                                                );
+                                            });
+                                }), Configs.getCPUCnt(), Configs.getCPUCnt());
     }
 
-    private Flux<OpenConnectionResponse> openConnectionsAndInitCachesInternal(
+    private Flux<ImmutablePair<ImmutablePair<String, DocumentCollection>, AddressInformation>> resolveAddressesPerCollection(
+            String containerLink,
             DocumentCollection collection,
             List<PartitionKeyRangeIdentity> partitionKeyRangeIdentities,
-            CosmosContainerProactiveInitConfig proactiveContainerInitConfig
+            CosmosContainerProactiveInitConfig proactiveContainerInitConfig,
+            AsyncDocumentClient.OpenConnectionAggressivenessHint hint
     ) {
+        if (proactiveContainerInitConfig.getProactiveConnectionRegionsCount() > 0) {
+            return Flux.fromStream(this.endpointManager.getReadEndpoints().stream())
+                    .take(proactiveContainerInitConfig.getProactiveConnectionRegionsCount())
+                    .flatMap(readEndpoint -> {
+                        if (this.addressCacheByEndpoint.containsKey(readEndpoint)) {
+                            return this.addressCacheByEndpoint.get(readEndpoint)
+                                    .addressCache
+                                    .resolveAddressesAndInitCaches(
+                                            containerLink,
+                                            collection,
+                                            partitionKeyRangeIdentities,
+                                            hint
+                                    );
+                        }
+                        return Flux.empty();
+                    });
+        }
+
+        return Flux.empty();
+    }
 
+    private Flux<Void> submitOpenConnectionInternal(
+            AddressInformation address,
+            DocumentCollection documentCollection,
+            CosmosContainerProactiveInitConfig proactiveContainerInitConfig,
+            int connectionPerEndpointCount,
+            AsyncDocumentClient.OpenConnectionAggressivenessHint hint
```

Comment:
since the hint is not used, so please remove it and all the upstream caller methods